diff --git a/node_modules/@types/lodash/common/object.d.ts b/node_modules/@types/lodash/common/object.d.ts
index 69552b0..7e88e0c 100644
--- a/node_modules/@types/lodash/common/object.d.ts
+++ b/node_modules/@types/lodash/common/object.d.ts
@@ -1021,17 +1021,19 @@ declare module "../index" {
         functionsIn(): CollectionChain<string>;
     }
 
-    type GetIndexedField<T, K> = K extends keyof T
-        ? T[K]
-        : K extends `${number}`
-            ? 'length' extends keyof T
-                ? number extends T['length']
-                    ? number extends keyof T
-                        ? T[number]
+    type GetIndexedField<T, K> = K extends string | number | bigint | boolean | null | undefined
+        ? `${K}` extends keyof T
+            ? T[`${K}`]
+            : K extends `${number}`
+                ? 'length' extends keyof T
+                    ? number extends T['length']
+                        ? number extends keyof T
+                            ? T[number]
+                            : undefined
                         : undefined
                     : undefined
                 : undefined
-            : undefined;
+        : undefined;
 
     type FieldWithPossiblyUndefined<T, Key> =
         | GetFieldType<Exclude<T, undefined>, Key>
@@ -1051,9 +1053,11 @@ declare module "../index" {
                 : undefined
         : P extends keyof T
             ? T[P]
-            : P extends `${infer FieldKey}[${infer IndexKey}]`
+            : P extends `${infer FieldKey}[${infer IndexKey}]${infer Rest}`
                 ? FieldKey extends keyof T
-                    ? IndexedFieldWithPossiblyUndefined<T[FieldKey], IndexKey>
+                    ? '' extends Rest
+                        ? IndexedFieldWithPossiblyUndefined<T[FieldKey], IndexKey>
+                        : FieldWithPossiblyUndefined<IndexedFieldWithPossiblyUndefined<T[FieldKey], IndexKey>, Rest>
                     : undefined
                 : IndexedFieldWithPossiblyUndefined<T, P>;
 
